packages/generator/src/generator.ts

```diff
@@ -55,31 +57,46 @@ const logger = createLogger({
   messageContext: 'generator'
 });
 
-export async function generate(
-  options: GeneratorOptions
-): Promise<void | EmitResult[]> {
+export async function generate(options: GeneratorOptions): Promise<void> {
   const project = await generateProject(options);
   if (!project) {
     throw Error('The project is undefined.');
   }
 
   await project.save();
-
   if (options.generateJs) {
-    project
+    const directories = project
       .getDirectories()
-      .filter(dir => !!dir.getSourceFile('tsconfig.json'))
-      .forEach(dir => {
-        logger.info(`Transpiling files in the directory: ${dir.getPath()}.`);
-
-        execa('tsc', { cwd: dir.getPath() }).catch(err => {
-          logger.error(`Error: Failed to generate js files: ${err}`);
-          process.exit(1);
-        });
-      });
+      .filter(dir => !!dir.getSourceFile('tsconfig.json'));
+    for (const chunk of splitInChunks(
+      directories,
+      options.processesJsGeneration || defaultValueProcessesJsGeneration
```

Comment:
This should not be necessary. Either it was overwritten or it is using the default, right?
```suggestion
      options.processesJsGeneration
```

Reply:
That would also mean you can get rid of the exported `defaultValueProcessesJsGeneration` variable.

Reply:
If somebody uses the `generate` method directly without the CLI the default value is not set form the command line parser.

Reply:
True that should not be the case but as long as it is, go for it ;)